.github/actions/patch-git/action.yml

```diff
@@ -0,0 +1,86 @@
+# Merge a "patch branch"
+name: 'Patch Git'
+description: 'Checkout the main and optional patch repositories and apply necessary patches'
+inputs:
+  name:
+    description: 'Human readable name of the project/product being checked out and patched'
+    required: true
+  local-dir:
+    description: 'Local workspace directory'
+    required: true
+  main-repository:
+    description: 'Main github repository in the form owner/repo'
+    required: true
+  main-branch:
+    description: 'Name of the main/master branch, leave empty to use the Git reference of the workflow run'
+    required: false
+  patch-repository:
+    description: 'Patch github repository in the form owner/repo'
+    required: false
+  patch-branch:
+    description: 'Name of the branch in patch-repository that has the patch commits'
+    required: false
+  work-branch:
+    description: 'Local work directory for the project/product'
+    required: true
+runs:
+  using: "composite"
+  steps:
+
+    - name: Checkout ${{ inputs.name }} repo
+      uses: actions/checkout@v3
+      with:
+        repository: ${{ inputs.main-repository }}
+        ref: ${{ inputs.main-branch }}
+        path: ${{ inputs.local-dir }}
+        # Need the full history for the merge below
+        fetch-depth: 0
```

Comment:
"Full history" but fetch depths of `0`?.. How does that work together?

Reply:
How about `--unshallow`?

Reply:
See `https://github.com/actions/checkout`: ` 0 indicates all history for all branches and tags.`